cockroachdb.go

```diff
@@ -0,0 +1,226 @@
+package testfixtures
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+const fkName string = "FOREIGN KEY"
+
+type cockroachDB struct {
+	baseHelper
+
+	skipResetSequences bool
+	resetSequencesTo   int64
+
+	constraints []crdbConstraint
+
+	tables    []string
+	sequences []string
+}
+
+type crdbConstraint struct {
+	tableName      string
+	constraintName string
+	constraintType string
+	details        string
+	validated      bool
+}
+
+func (h *cockroachDB) init(db *sql.DB) error {
+	var err error
+
+	h.tables, err = h.tableNames(db)
+	if err != nil {
+		return err
+	}
+
+	h.sequences, err = h.getSequences(db)
+	if err != nil {
+		return err
+	}
+
+	h.constraints, err = h.getConstraints(db)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (*cockroachDB) paramType() int {
+	return paramTypeDollar
+}
+
+func (*cockroachDB) databaseName(q queryable) (string, error) {
+	var dbName string
+	err := q.QueryRow("SELECT current_database()").Scan(&dbName)
+	return dbName, err
+}
+
+func (h *cockroachDB) tableNames(q queryable) ([]string, error) {
+	var tables []string
+
+	sql := `
+	        SELECT pg_namespace.nspname || '.' || pg_class.relname
+		FROM pg_class
+		INNER JOIN pg_namespace ON pg_namespace.oid = pg_class.relnamespace
+		WHERE pg_class.relkind = 'r'
+		  AND pg_namespace.nspname NOT IN ('pg_catalog', 'information_schema')
+		  AND pg_namespace.nspname NOT LIKE 'pg_toast%'
+		  AND pg_namespace.nspname NOT LIKE 'crdb_internal%'
```

Comment:
This line could be moved to the `postgresql.go` file

Reply:
Hi @andreynering,

Thanks for reviewing this PR, this cannot be used as it is with pg since it uses `SHOW CONSTRAINTS FROM` which is crdb specific to get all constraints for a table. To have common impl between pg and crdb we will probably need to use pg_catalog to get constraints. `SHOW CONSTRAINTS FROM` is more convenient for the impl since it provides textual description of the constraint that can be used directly to re-create it, which in case of pg_catalog is not there and SQL have to be constructed from low level info.

Reply:
After quick research I think it should be easy to do, I will explore more and open a new PR once implemented.